streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -614,6 +615,12 @@ public StreamThread(final Time time,
             streamsMetrics,
             time.milliseconds()
         );
+        ThreadMetrics.addThreadStateTelemetryMetric(threadId,
+                streamsMetrics,
+                (metricConfig, now) -> this.state().ordinal());
+        ThreadMetrics.addThreadStateMetric(threadId,
+                streamsMetrics,
+                (metricConfig, now) -> this.state().name().toLowerCase(Locale.getDefault()));
```

Comment:
For the corresponding client metric we just use
```
 ClientMetrics.addStateMetric(streamsMetrics, (metricsConfig, now) -> state);
 ```
 
 Why so "complicated" -- I am also ok to update the code for the client metric. But both should be the same?
 
 (Or maybe keep `... -> state` and add a "fancy" `toString()` overload to both `enum` (for client and thread) which model the state?

Reply:
The KIP specifies lower-case string but I like the other approach so I'll update it.